### PR TITLE
Update docstrings for backward methods

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

After recent changes in #13863, I forgot to update the docstrings.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @rohitgr7 @awaelchli @akihironitta

### CODE REVIEW

File: src/pytorch_lightning/plugins/precision/deepspeed.py [hunk 1]
```diff
@@ -72,6 +72,16 @@ def backward(
...
     ) -> None:
+        """Performs back-propagatin using DeepSpeed's engine.
+
+        Args:
+            model: the model to be optimized
+            closure_loss: the loss tensor
+            optimizer: ignored for DeepSpeed
+            optimizer_idx: ignored for DeepSpeed
+            *args: additional positional arguments for the :meth:`deepspeed.DeepSpeedEngine.backward` call
+            *kwargs: additional keyword arguments for the :meth:`deepspeed.DeepSpeedEngine.backward` call
```
Comment (rohitgr7) [inline src/pytorch_lightning/plugins/precision/deepspeed.py:83]:
```suggestion
            \*args: additional positional arguments for the :meth:`deepspeed.DeepSpeedEngine.backward` call
            \**kwargs: additional keyword arguments for the :meth:`deepspeed.DeepSpeedEngine.backward` call
```

File: src/pytorch_lightning/plugins/precision/precision_plugin.py [hunk 2]
```diff
@@ -74,6 +74,9 @@ def backward(
             model: the model to be optimized
             closure_loss: the loss value obtained from the closure
             optimizer: current optimizer being used. ``None`` if using manual optimization
+            *args: Positional arguments intended for the actual function that performs the backward, like
+                :meth:`~torch.Tensor.backward`.
+            *kwargs: Keyword arguments for the same purpose as `*args`.
```
Comment (rohitgr7) [inline src/pytorch_lightning/plugins/precision/precision_plugin.py:79]:
```suggestion
            \*args: Positional arguments intended for the actual function that performs the backward, like
                :meth:`~torch.Tensor.backward`.
            \**kwargs: Keyword arguments for the same purpose as ``*args``.
```

File: src/pytorch_lightning/strategies/strategy.py [hunk 3]
```diff
@@ -183,6 +183,11 @@ def backward(
 
         Args:
             closure_loss: a tensor holding the loss value to backpropagate
+            optimizer: An optional optimizer that gets passed down to the precision plugin's backward
+            optimizer_idx: An optional optimizer index that gets passed down to the precision plugin's backward
+            *args: Positional arguments that get passed down to the precision plugin's backward, intended as arguments
+                for the actual function that performs the backward, like :meth:`~torch.Tensor.backward`.
+            *kwargs: Keyword arguments for the same purpose as `*args`.
```
Comment (rohitgr7) [inline src/pytorch_lightning/strategies/strategy.py:190]:
```suggestion
            \*args: Positional arguments that get passed down to the precision plugin's backward, intended as arguments
                for the actual function that performs the backward, like :meth:`~torch.Tensor.backward`.
            \**kwargs: Keyword arguments for the same purpose as ``*args``.
```

File: src/pytorch_lightning/plugins/precision/precision_plugin.py [hunk 4]
```diff
@@ -74,6 +74,9 @@ def backward(
             model: the model to be optimized
             closure_loss: the loss value obtained from the closure
             optimizer: current optimizer being used. ``None`` if using manual optimization
```
Comment (rohitgr7) [inline src/pytorch_lightning/plugins/precision/precision_plugin.py:76]: also optimizer_idx

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Comment by kaushikb11** [comment]
@tchaton @justusschock

**Review by akihironitta — APPROVED** [review]

File: src/pytorch_lightning/plugins/precision/deepspeed.py [hunk 5]
```diff
@@ -72,6 +72,16 @@ def backward(
         *args: Any,
         **kwargs: Any,
     ) -> None:
+        r"""Performs back-propagatin using DeepSpeed's engine.
```
Comment (tchaton) [inline src/pytorch_lightning/plugins/precision/deepspeed.py:75]:
```suggestion
        r"""Performs back-propagation using DeepSpeed's engine.
```

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**